### PR TITLE
SPCTR-3256 Fix: Taxonomy List - Resolved an issue wherein the hierarchy would not be displayed as intended in the list layout.

### PR DESCRIPTION
**Pull request by AbhijitNage123**
### Description

- https://brainstormforce.atlassian.net/browse/SPCTR-3256

### Screenshots

- https://d.pr/v/OBW409

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### How has this been tested?

- Added to the card

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: blocks-config/taxonomy-list/class-uagb-taxonomy-list.php [hunk 1]
```diff
@@ -594,6 +594,69 @@ public function get_link_of_individual_categories( $slug, $taxonomy_type ) {
...
+		 * @param string $seperatorStyle of separator style.
+		 * @param string $title_tag of title tag.
+		 * @param bool   $show_hierarchy of show hierarchy.
+		 * @since x.x.x
+		 * @return void
+		 */
+		public function display_terms_recursive( $term, $taxonomy_type, $showCount, $seperatorStyle, $title_tag, $show_hierarchy ) {
+			if ( $term instanceof WP_Term ) {
+				$child_link = $this->get_link_of_individual_categories( $term->slug, $taxonomy_type );
+				echo wp_kses_post(
```
Comment (rahulvarma722) [inline blocks-config/taxonomy-list/class-uagb-taxonomy-list.php:641]: Please use srintf for clarity and avoid concatenation

File: blocks-config/taxonomy-list/class-uagb-taxonomy-list.php [hunk 2]
```diff
@@ -594,6 +594,69 @@ public function get_link_of_individual_categories( $slug, $taxonomy_type ) {
...
+					'<' . esc_html( $title_tag ) . ' class="uagb-tax-link-wrap"><a class="uagb-tax-link" href="' . esc_url( $child_link ) . '">' . esc_html( $term->name ) . '</a>' . 
+					( $showCount ? ' (' . esc_attr( $term->count ) . ') ' : '' ) . 
+					'<' . esc_html( $title_tag ) . '>' . 
+					( 'none' !== $seperatorStyle ? '<div class="uagb-tax-separator"></div>' : '' ) . 
+					'</li>'
+				);
+
+				if ( $show_hierarchy && ! empty( $term->children ) && is_array( $term->children ) ) {
+					foreach ( $term->children as $child ) {
+						echo '<ul class="uagb-taxonomy-list-children">';
```
Comment (rahulvarma722) [inline blocks-config/taxonomy-list/class-uagb-taxonomy-list.php:652]: use sprintf

File: src/blocks/taxonomy-list/render.js [hunk 3]
```diff
@@ -32,6 +36,74 @@ const Render = ( props ) => {
 	} else if ( 'list' === layout ) {
 		Tag = titleTag ? titleTag : 'div';
 	}
+	const query = { per_page: -1, hide_empty: ! showEmptyTaxonomy, context: 'view' };
```
Comment (rahulvarma722) [inline src/blocks/taxonomy-list/render.js:39]: Keep these things in a separate file, such as utils. In the render file, only visual elements should be included.

File: blocks-config/taxonomy-list/class-uagb-taxonomy-list.php (hunk 1 above)
Comment (AbhijitNage123) [reply blocks-config/taxonomy-list/class-uagb-taxonomy-list.php]: Done

File: blocks-config/taxonomy-list/class-uagb-taxonomy-list.php (hunk 2 above)
Comment (AbhijitNage123) [reply blocks-config/taxonomy-list/class-uagb-taxonomy-list.php]: Done

File: blocks-config/taxonomy-list/class-uagb-taxonomy-list.php [hunk 4]
```diff
@@ -594,6 +594,69 @@ public function get_link_of_individual_categories( $slug, $taxonomy_type ) {
...
+					esc_html( $term->name ),
+					( boolval( $showCount ) ? ' (' . esc_attr( $term->count ) . ') ' : '' ),
+					esc_html( $title_tag )
+				);
+
+				if ( $show_hierarchy && ! empty( $term->children ) && is_array( $term->children ) ) {
+					foreach ( $term->children as $child ) {
+						echo sprintf( '<ul class="uagb-taxonomy-list-children">' );
+						$this->display_terms_recursive( $child, $taxonomy_type, $showCount, $seperatorStyle, $title_tag, $show_hierarchy );
+						echo sprintf( '</ul>' );
```
Comment (rahulvarma722) [inline blocks-config/taxonomy-list/class-uagb-taxonomy-list.php:654]:
```suggestion
				echo sprintf( '<ul class="uagb-taxonomy-list-children">%s</ul>', $this->display_terms_recursive( $child, $taxonomy_type, $showCount, $seperatorStyle, $title_tag, $show_hierarchy ) );
```

File: src/blocks/taxonomy-list/render.js [hunk 5]
```diff
@@ -21,17 +24,88 @@ const Render = ( props ) => {
...
+						<a className="uagb-tax-link" href={link} target="_blank" rel="noreferrer noopener">
+							{ renderCategoryName( name ) }
+						</a>
+						{showCount && ` (${count})`}
+					</Tag>
+					{'none' !== seperatorStyle && <div className="uagb-tax-separator" ></div>}
+				</li>
+			);
+		} 
+		// if ( taxonomyType === 'post_tag' )
```
Comment (rahulvarma722) [inline src/blocks/taxonomy-list/render.js:87]: Remove commented code.

File: blocks-config/taxonomy-list/class-uagb-taxonomy-list.php (hunk 4 above)
Comment (rahulvarma722) [reply blocks-config/taxonomy-list/class-uagb-taxonomy-list.php:654]: Please make sure this will give the correct HTML structure.

Comment (rahulvarma722) [reply blocks-config/taxonomy-list/class-uagb-taxonomy-list.php:654]: Ok this is giving direct HTML which is already echo we can keep that at is is

File: src/blocks/taxonomy-list/render.js (hunk 5 above)
Comment (AbhijitNage123) [reply src/blocks/taxonomy-list/render.js]: Done

**Review by rahulvarma722 — APPROVED** [review]

**Review by adi3890 — APPROVED** [review]

**Review by adi3890 — APPROVED** [review]

**Pull request merged**